### ISSUE TITLE
Bug: no-unused-expressions errors for function calls in ternaries but shouldn't

### ISSUE THREAD

**Issue by gk-bbai** [eslint/eslint#19083]
### Environment

Node version: Using ESLint Playground
npm version: Using ESLint Playground
Local ESLint version: 9.13.0
Global ESLint version: 9.13.0
Operating System: Using ESLint Playground

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
/*eslint no-unused-expressions: "error"*/
```
</details>

```js
let i = 0;
function foo() {
    i = 1
}

function bar() {
    i = 2
}

const x = Math.random() > 0.5

x ? foo() : bar()

console.log(i)
```

### What did you expect to happen?

The below statement from the rule documentation:

> This rule does not apply to function calls or constructor calls with the new operator, because they could have side effects on the state of the program.

should apply to ternary expressions by default, resulting in no error in the above code.

I don't think users should have to set `allowTernary: true` for this case (the case where ternary operands include function calls, as per the quote from the documentation above).

### What actually happened?

> 12:1 Expected an assignment or function call and instead saw an expression.  ([no-unused-expressions](https://eslint.org/docs/rules/no-unused-expressions))

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLWV4cHJlc3Npb25zOiBcImVycm9yXCIqL1xuXG5sZXQgaSA9IDA7XG5mdW5jdGlvbiBmb28oKSB7XG4gICAgaSA9IDFcbn1cblxuZnVuY3Rpb24gYmFyKCkge1xuICAgIGkgPSAyXG59XG5cbmNvbnN0IHggPSBNYXRoLnJhbmRvbSgpID4gMC41XG5cbnggPyBmb28oKSA6IGJhcigpXG5cbmNvbnNvbGUubG9nKGkpXG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @gk-bbai, thanks for the issue!

This works as intended because the value of `x ? foo() : bar()` is unused. In a codebase that doesn't allow this pattern as an alternative to if-else statements, this is a possible error (e.g., the code should have been `baz(x ? foo() : bar())` or `baz = x ? foo() : bar()`). In a codebase that allows this pattern, you can set `allowTernary: true` option.
